### PR TITLE
compileopts: move {root} replacement to compileopts.Emulator()

### PR DESCRIPTION
**Pull request by dgryski**
Follow up for https://github.com/tinygo-org/tinygo/pull/2387#issuecomment-1002309218

### CODE REVIEW

**Comment by dgryski** [comment]
I think I know why CI failed.

**Comment by deadprogram** [comment]
@dgryski please note merge conflict needing resolution, please.

**Comment by dgryski** [comment]
Fixed.

File: compileopts/config.go [hunk 1]
```diff
@@ -462,6 +462,15 @@ func (c *Config) WasmAbi() string {
 	return c.Target.WasmAbi
 }
 
+// Emulator returns the emulator target config
+func (c *Config) Emulator() []string {
+	var emulator []string
+	for _, s := range c.Target.Emulator {
+		emulator = append(emulator, strings.ReplaceAll(s, "{root}", goenv.Get("TINYGOROOT")))
+	}
```
Comment (aykevl) [inline compileopts/config.go:471]: This works, but a more efficient version is the following:

```suggestion
	emulator = make([]string, len(c.Target.Emulator))
	for i, s := range c.Target.Emulator {
		emulator[i] = strings.ReplaceAll(s, "{root}", goenv.Get("TINYGOROOT"))
	}
```

...probably not worth the effort though. Just an idea. It's even possible the Go compiler recognizes the idiom and optimizes it.

File: main_test.go [hunk 2]
```diff
@@ -369,13 +369,20 @@ func runTestWithConfig(name string, t *testing.T, options compileopts.Options, c
 
 	// Create the test command, taking care of emulators etc.
 	var cmd *exec.Cmd
-	if len(spec.Emulator) == 0 {
+
+	// Probably not the best place to handle this
+	var emulator []string
+	for _, s := range spec.Emulator {
+		emulator = append(emulator, strings.ReplaceAll(s, "{root}", goenv.Get("TINYGOROOT")))
+	}
```
Comment (aykevl) [inline main_test.go]: Yeah, that kind of duplication doesn't look good.
What about this?

```suggestion
	config := compileopts.Config{Target: spec}
	emulator := config.Emulator()
```

A bit hacky but it works.
Also, `spec.Emulator` is also used below. It probably doesn't matter much though.

Comment (dgryski) [reply main_test.go]: Fixed and repushed.

**Comment by deadprogram** [comment]
@dgryski can you please resolve merge conflicts? Thank you.

**Comment by dgryski** [comment]
Rebased.

**Comment by deadprogram** [comment]
Thanks for the improvement @dgryski now merging.

**Pull request merged**